rails_programming/assets_and_navigation/turbo_drive.md

```diff
@@ -0,0 +1,153 @@
+
+Restoration visits are visits with the action of *restore*. This is used by Turbo Drive internally and you **should not** annotate a link with an action of restore.
+
+#### HTTP Request Methods
+
+By default, link clicks are sent with `GET` requests. However, Turbo Drive will scan `<a>` tags in your application for the `turbo-method` attribute to override the `GET` action. 
+
+For instance:
+
+~~~erb
+<%= link_to "Delete Article", article_path(@article), data: { turbo_method: "delete" }  %>
+~~~
+
+which will generate:
+
+~~~html
+<a href="..." data-turbo-method="delete">Delete Article</a>
+~~~
+
+This creates a link that will use the `DELETE` method. However, it is suggested that you use a button or form for anything that isn't a `GET` request
+
+#### Disable Turbo Drive
+
+There will be times where you do not want Turbo Drive to control navigation and want a full page reset.
+
+You can disable Turbo Drive by adding `data-turbo="false` directly on your links or on the parent containing them.
+
+For instance:
+
+~~~erb
+<div data-turbo="false">
+  <%= link_to "foo", "bar" %>
+  <%= link_to "baz", "qux", data: { turbo: "true" } %>
+</div>
+~~~
+
+In the above example, we created a parent div with `data-turbo="false"`, which disables Turbo Drive on all elements inside of it. However, we also added `data: { turbo: "true" }` to the second `link_to`. This will turn Turbo Drive back on for that particular element.
+
+This showcases that you can disable and re-enable Turbo Drive in regions of your website and select specific elements to enable or disable it on. Remember that when using ERB tags (like `link_to`) we use the syntax style of `data: { foo: "bar" }` rather than the HTML style of `data-foo="bar"`
+
+### Forms
+
+We cover Rails forms much deeper later, but because they are entwined with Turbo Drive we do need to cover some common information that will come together in the forms lessons that follow.
+
+If you don't understand some of the information as it is explained below just wait until the form basics lesson and it should all come together.
+
+#### Form Submissions
+
+While most link requests are issued as GET requests, forms are different. You often submit forms as a POST, PATCH or DELETE request in order to manipulate some data that lives on the server.
+
+Turbo Drive intercepts all Form Submissions by default and submits them itself to the server on your behalf. What you must keep in mind going forwards into the form lessons is that when Turbo expects the server usually to return an HTTP status of HTTP 303. This is a redirect response. Think of it like this, after you submit a form usually you want to do be directed somewhere afterwards. That could be back to the same page with a blank form ready for another submission, or it could be to view the resource you just created with the form, but you expect to be redirected away from the current page with the completed form on it. There are two exceptions to this expectation
```

Comment:
```suggestion
Turbo Drive intercepts all Form Submissions by default and submits them to the server on your behalf. As we advance into the form lessons, you must remember that Turbo expects the server to return an HTTP status of HTTP 303 or, in other words, a redirect.

Think of it like this: after you submit a form, you usually want to be directed somewhere afterwards. That could be back to the same page with a blank form ready for another submission, or it could be to view the resource you just created with the form, but you expect to be redirected away from the current page with the completed form on it. There are two exceptions to this expectation.
```